### PR TITLE
support initial search in `PAGER_BUILTIN`

### PR DESCRIPTION
**Pull request by ThomasBreuer**
When `PageSource` calls `Pager` with the data for a kernel function, it specifies the start line by a search string.
This works well for external pagers such as `less`, but up to now `PAGER_BUILTIN` did not honor this hint.

With the changes proposed here,
calls such as `PageSource( IS_SSORT_LIST_DEFAULT )` will work as expected also when `PAGER_BUILTIN` is used.

(The code is a hack, because GAP does not support regular expressions and because the search in C code files is based only on some heuristics.)

### CODE REVIEW

File: lib/pager.gi [hunk 1]
```diff
@@ -136,6 +129,31 @@ BindGlobal("PAGER_BUILTIN", function( lines )
...
+        nam := r.start{[7..Length(r.start)]};
+        linepos:= PositionsProperty(lines,
+                    l -> PositionSublist(l, nam) <> fail);
+        linepos:= First(linepos,
+                    i -> PositionSublist(lines[i], "Obj")
+                         < PositionSublist(lines[i], nam) and
+                         ForAll(lines[i]{[PositionSublist(lines[i], "Obj")+3
+                                   .. PositionSublist(lines[i], nam)-1]},
+                           x -> x = ' '));
+      fi;
```
Comment (fingolfin) [inline lib/pager.gi:150]: Should we issue an error when `r.start` is unsupported?
```suggestion
      else
        Error("unsupported r.start value: ", r.start);
      fi;
```

But I am also OK with ignoring it

**Review by fingolfin — APPROVED** [review]

**Comment by ThomasBreuer** [comment]
@frankluebeck suggested that it is better to support an initial search in general for the builtin pager, but to drop the support for a regular expression. I will update the pull request.

**Comment by frankluebeck** [comment]
Sorry, I communicated my review in person. I should have left a short note here as well.

File: lib/pager.gi [hunk 2]
```diff
@@ -101,29 +101,50 @@ GAPInfo.UserPreferences.Pager := UserPreference("Pager");
...
+##    <K>true</K> or <K>false</K> (default <K>false</K>),
+##    <K>true</K> means that lines not fitting into one terminal line
+##    (see <Ref Func="SizeScreen"/>) will be distributed to several lines,
+##  </Item>
+##  <Mark>start</Mark>
+##  <Item>
+##    a positive integer denoting the position of the first line that is
+##    shown,
+##    or a string starting with <C>"/"</C>, meaning that the first line
+##    containing the substring behind <C>"/"</C> shall be shown;
```
[the remark this answers is not in the document]
Comment (ThomasBreuer) [reply lib/pager.gi:121]: The point is that `PageSource` prescribes the search string and just calls `Pager`, without distinguishing whether `PAGER_BUILTIN` or an external pager will be used.

[the remark this answers is not in the document]
Comment (fingolfin) [reply lib/pager.gi:121]: But the documentation of `Pager` says nothing about this either. I don't mind so much if some internal functions work that way, but if this now gets documented it feels odd to mimic the behavior of `less` and `more` here -- why not just pass the search string, and then let the code calling the external pager prepend the `/`? Just like it prepends `+` for line numbers?

In the end I can live with it either way, as long as it is a deliberate choice.

**Comment by frankluebeck** [comment]
So far we are using `less` with `+[initial command]` syntax. The initial command can be anything, e.g., a search like `/xyz` or going to the end `G` or ... or going to line 13 `13` or `13g`.

**Review by fingolfin — APPROVED** [review]

**Comment by ThomasBreuer** [comment]
The test `codecov/project` is now running for more than four days. (Started 2024-03-28 17:29:05.)
Apparently something went wrong.

**Comment by ThomasBreuer** [comment]
The test `codecov/project` is now running for more than 12 days.
How can this be stopped/restarted?

**Comment by fingolfin** [comment]
That the codecov test is reported as "running" doesn't mean something is running somewhere, just that received an event "test has started", but somehow the matching "test has finished" was not delivered (can always happen due to e.g. network issues).

It is also perfectly fine to ignore that test. Normally I would have, but in this case I decided to close&reopen to re-run CI anyway, because something else was weird (all `codecov/...` checks were listed twice). Strangely enough, this remains, but not for other PRs. Very odd.

**Pull request merged**